mautic/mautic
# Contact timezone saved to the contact from the page hit tracking

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | false
| New feature? | true
| Related user documentation PR URL | Do we need to document this?
| Related developer documentation PR URL | null
| Issues addressed (#s or URLs) | Preparation for https://github.com/mautic/mautic/issues/539
| BC breaks? | N
| Deprecations? | N

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
This PR adds a new Timezone custom field and fills it with a timezone in the PHP's timezone format (continent/city_or_country) based on what Fingerprint2 library sends us as timezone offset in the page hits.

If the Timezone field already exists among the contact custom fields, the migration will just update it.

#### Steps to test this PR:
1. Pull this PR
2. Run `app/console doctrine:migrations:migrate`
3. Check that the Timezone custom field was created or set as a core field if existed before
4. Visit a page tracked by Mautic JS tracking in a incognito browser window
5. Check that the contact who was just tracked is the correct timezone in the contact detail view as well as in the contact edit view.
6. You can also run `phpunit` in the app folder to test one unit test that was added.

This PR also fixes the timezone field type which stored only one part of the timezone previously.
